apisix/plugins/server-info.lua

```diff
@@ -158,25 +150,70 @@ local function report(premature, report_ttl)
         end
     end
 
-    server_info.last_report_time = ngx_time()
-
+    -- get inside etcd data, if not exist, create it
     local key = "/data_plane/server_info/" .. server_info.id
-    local ok, err = core.etcd.set(key, server_info, report_ttl)
+    local res, err = core.etcd.get(key)
+    if err ~= nil then
+        core.log.error("failed to get server_info from etcd: ", err)
+    end
+
+    if  not res.body.node then
+        local newres, err = core.etcd.set(key, server_info, report_ttl)
+        if not newres then
+            core.log.error("failed to set server_info: ", err)
+            return
+        end
+
+        -- set lease_id to ngx dict
+        local _, err = internal_status:set("lease_id", newres.body.lease_id)
+        if err ~= nil then
+            core.log.error("failed to save boot_time to shdict: ", err)
+        end
+
+        return
+    end
+
+    local ok = core.table.deep_eq(server_info, res.body.node.value)
+    -- not equal, update it
     if not ok then
-        core.log.error("failed to report server info to etcd: ", err)
+        local newres, err = core.etcd.set(key, server_info, report_ttl)
+        if not newres then
+            core.log.error("failed to set server_info to etcd: ", err)
+            return false, err
+        end
+
+        -- update lease_id
+        local _, err = internal_status:set("lease_id", res.body.lease_id)
+        if err ~= nil then
+            core.log.error("failed to set lease_id to shdict: ", err)
+        end
```

Comment:
This paragraph is same to the one in the `if  not res.body.node then .... end` block, we can create a function to reuse it.